File: Demo/Demo/Example5ViewController.m
//
//  Example5ViewController.m
//  Demo
//
//  Created by André Henrique Silva on 08/05/15.
//  Copyright (c) 2015 André Henrique Silva. All rights reserved.
//

#import "Example5ViewController.h"
#import "AHPagingMenuViewController.h"
#import "Example4ViewController.h"

@interface Example5ViewController ()

@property (nonatomic) NSInteger current;

@end

@implementation Example5ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    self.current = 1;
}

- (IBAction)newController:(id)sender {
    [self.pagingMenuViewController addNewController:[Example4ViewController new] andTitleView:[NSString stringWithFormat:@"New %ld", (long)self.current]];
    [self.pagingMenuViewController setPosition:[self.pagingMenuViewController.viewControllers count] -1 animated:YES];
    self.current++;
}

- (IBAction)goToFirst:(id)sender {
    [self.pagingMenuViewController setPosition:0 animated:YES];
}

- (IBAction)goToLast:(id)sender {
     [self.pagingMenuViewController setPosition:[self.pagingMenuViewController.viewControllers count] -1 animated:YES];
}

- (IBAction)setArrow:(id)sender {
    [self.pagingMenuViewController setShowArrow:!self.pagingMenuViewController.showArrow];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}
- (IBAction)next:(id)sender {
    [self.pagingMenuViewController goNextView];
}

- (IBAction)reset:(id)sender {
    [self.pagingMenuViewController setBounce:!self.pagingMenuViewController.isBounce];
}

- (IBAction)Previeus:(id)sender {
      [self.pagingMenuViewController goPrevieusView];
}

- (IBAction)changeFonte:(id)sender {
    [self.pagingMenuViewController setChangeFont:!self.pagingMenuViewController.changeFont];
}

- (IBAction)transform:(id)sender {
    [self.pagingMenuViewController setTransformScale: !self.pagingMenuViewController.transformScale];
}

- (IBAction)fade:(id)sender {
     [self.pagingMenuViewController setFade: !self.pagingMenuViewController.fade];
}

- (IBAction)randonFirst:(id)sender {
    [self.pagingMenuViewController setSelectedColor:[self randomColor]];
}

- (IBAction)randonSecond:(id)sender {
    [self.pagingMenuViewController setDissectedColor:[self randomColor]];
}

- (IBAction)navColor:(id)sender {
    [self.pagingMenuViewController setNavBackgroundColor:[self randomColor]];
}

- (IBAction)navLineColor:(id)sender
{
    [self.pagingMenuViewController setNavLineBackgroundColor:[self randomColor]];
}

- (IBAction)scaleTitle:(id)sender {
    [self.pagingMenuViewController setScaleMax:((double)arc4random() / 0x100000000) + 0.3 andScaleMin:((double)arc4random() / 0x100000000)];
}

- (IBAction)firstFont:(id)sender {
    [self.pagingMenuViewController setSelectedFont:[UIFont fontWithName:@"Chalkduster" size:16]];
}

- (IBAction)secondFont:(id)sender {
     [self.pagingMenuViewController setDissectedFont:[UIFont fontWithName:@"BradleyHandITCTT-Bold" size:16]];
}

-(UIColor *)randomColor
{
    CGFloat hue = ( arc4random() % 256 / 256.0 );  //  0.0 to 1.0
    CGFloat saturation = ( arc4random() % 128 / 256.0 ) + 0.5;  //  0.5 to 1.0, away from white
    CGFloat brightness = ( arc4random() % 128 / 256.0 ) + 0.5;  //  0.5 to 1.0, away from black
    UIColor *color = [UIColor colorWithHue:hue saturation:saturation brightness:brightness alpha:1];
    return color;
}


@end

File: Swift/Demo/Demo/Example5ViewController.swift
//
//  Example5ViewController.swift
//  Demo
//
//  Created by André Henrique Silva on 18/05/15.
//  Copyright (c) 2015 André Henrique Silva. All rights reserved.
//

import UIKit

class Example5ViewController: UIViewController {
    
    var current = 1;
    
    @IBAction func randonNavColor(_ sender: AnyObject) {
    
    }

    @IBAction func randNavColorMain(_ sender: AnyObject) {
        self.pagingMenuViewController().setChangeColor(!self.pagingMenuViewController().changeFont)
    }
 
    @IBAction func changeFont(_ sender: AnyObject) {
        self.pagingMenuViewController().setChangeFont(!self.pagingMenuViewController().changeFont)
    }

    @IBAction func changeFace(_ sender: AnyObject) {
        self.pagingMenuViewController().setFade(!self.pagingMenuViewController().fade)
    }
    
    @IBAction func changeTransform(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setTransformScale(!self.pagingMenuViewController().transformScale);
    }
    
    @IBAction func changeArrow(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setShowArrow(!self.pagingMenuViewController().showArrow)
    }
    
    @IBAction func goFirst(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setPosition(0 , animated: true)
    
    }

    @IBAction func addController(_ sender: AnyObject)
    {
        let newController = ExampleViewController()
        newController.view.backgroundColor = self.getRandomColor();
        self.pagingMenuViewController().addNewController(newController, title: "New \(current)" as AnyObject)
        self.pagingMenuViewController().setPosition(self.pagingMenuViewController().viewControllers!.count - 1, animated: true)
        self.current += 1;
    }
    
    @IBAction func goLast(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setPosition(self.pagingMenuViewController().viewControllers!.count - 1, animated: true)
    
    }
    
    @IBAction func goNext(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setPosition(self.pagingMenuViewController().currentPage + 1, animated: true)
    }
    
    @IBAction func goPrevius(_ sender: AnyObject)
    {
         self.pagingMenuViewController().setPosition(self.pagingMenuViewController().currentPage - 1, animated: true)
    }
    
    @IBAction func changeSecondFont(_ sender: AnyObject)
    {
        let font = UIFont.init(name: "Chalkduster", size: CGFloat(16))
        self.pagingMenuViewController().setDissectFont(font!)

    }
    
    @IBAction func changeMainFont(_ sender: AnyObject)
    {
        let font = UIFont.init(name: "BradleyHandITCTT-Bold", size: CGFloat(16))
        self.pagingMenuViewController().setSelectFont(font!)
    }
    
    @IBAction func randonNavTitleTransform(_ sender: AnyObject)
    {
        let scale = CGFloat(arc4random_uniform(10)) / 10.0
        self.pagingMenuViewController().setScaleMax(scale + 0.5, scaleMin: scale)
    }

    @IBAction func randonSecondColor(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setDissectColor(self.getRandomColor())
    }
    
    @IBAction func randonFirstColor(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setSelectColor(self.getRandomColor())
    }
    
    @IBAction func changeBounce(_ sender: AnyObject)
    {
        self.pagingMenuViewController().setBounce(!self.pagingMenuViewController().bounce)
    }
    
    func getRandomColor() -> UIColor{
        
        let randomRed:CGFloat = CGFloat(drand48())
        let randomGreen:CGFloat = CGFloat(drand48())
        let randomBlue:CGFloat = CGFloat(drand48())
        return UIColor(red: randomRed, green: randomGreen, blue: randomBlue, alpha: 1.0)
        
    }
}
